opm/simulators/wells/WellGroupHelpers.hpp

```diff
@@ -446,6 +496,48 @@ namespace Opm {
         return;
     }
 
+    inline double groupFractionFromPotentials(const Group& group, const Schedule& schedule, const WellStateFullyImplicitBlackoil& wellState, const int reportStepIdx, const int phasePos, const bool isInjector) {
+        double groupTotalGuideRate = 0.0;
+        const Group& groupParent = schedule.getGroup(group.parent(), reportStepIdx);
+        for (const std::string& groupName : groupParent.groups()) {
+            const Group& groupTmp = schedule.getGroup(groupName, reportStepIdx);
+
+            // only count group under group control from its parent
+            if (isInjector) {
+                const Group::InjectionCMode& currentGroupControl = wellState.currentInjectionGroupControl(groupName);
+                if (currentGroupControl != Group::InjectionCMode::FLD)
+                    continue;
+            } else {
+                const Group::ProductionCMode& currentGroupControl = wellState.currentProductionGroupControl(groupName);
+                if (currentGroupControl != Group::ProductionCMode::FLD)
+                    continue;
+            }
+
+            if ( (groupTmp.isProductionGroup() && !isInjector)) {
+                assert(false);
```

Comment:
Should this not be something like
```
groupTotalGuideRate += wellState.currentGroupProductionPotentials(groupName)[phasePos];
```

Reply:
The guideRate object takes care of the productionPotential